user_app/views.py

```diff
@@ -1,3 +1,41 @@
-from django.shortcuts import render
-
+from django.shortcuts import render,get_object_or_404
+from rest_framework.views import APIView
+from .models import User
+from rest_framework.response import Response
+from rest_framework.authtoken.models import Token
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from django.contrib.auth import authenticate
+from rest_framework.status import HTTP_200_OK,HTTP_204_NO_CONTENT,HTTP_400_BAD_REQUEST,HTTP_404_NOT_FOUND,HTTP_201_CREATED
 # Create your views here.
+
+# use cookies, grab user through cookies
+class Sign_up(APIView):
+    def post(self,request):
+        email = request.data.get('email')
+        password = request.data.get('password')
+        username = email
+        new_user = User.objects.create_user(username=username,password=password,email=email)
+        token = Token.objects.create(user=new_user)
+        # change when making front end
+        return Response({'user':new_user.email,'token':token.key},status=HTTP_201_CREATED)
+
+class Log_in(APIView):
+    def post(self,request):
+        username = request.data.get('email')
+        password = request.data.get('password')
+        user = authenticate(username=username,password=password)
+        if user:
+            token, created = Token.objects.get_or_create(user = user)
```

Comment:
Please add some error boundaries to your views otherwise a user will continue to see a 500 server error if they're sending an improper request. Instead they should see a 400 which lets the user know the request they sent is missing information